app/services/cancel_reservation.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class CancelReservation
+  def initialize(reservation)
+    @reservation = reservation
+  end
+
+  def call
+    return false unless booked?
+
+    reservation.update(status: :canceled)
```

Comment:
```suggestion
    reservation.canceled!
```

Same thing as above, it is just a different way to write this if this is enum value